src/shared/base-filesystem.c

```diff
@@ -38,25 +38,47 @@ static const BaseFilesystem table[] = {
         { "proc",  0755, NULL,                         NULL, true },
         { "sys",   0755, NULL,                         NULL, true },
         { "dev",   0755, NULL,                         NULL, true },
-#if defined(__i386__) || defined(__x86_64__)
+
         /* Various architecture ABIs define the path to the dynamic loader via the /lib64/ subdirectory of
          * the root directory. When booting from an otherwise empty root file system (where only /usr/ has
          * been mounted into) it is thus necessary to create a symlink pointing to the right subdirectory of
          * /usr/ first — otherwise we couldn't invoke any dynamic binary. Let's detect this case here, and
          * create the symlink as needed should it be missing. We prefer doing this consistently with Debian's
          * multiarch logic, but support Fedora-style multilib too.*/
+#if defined(__aarch64__)
+#elif defined(__alpha__)
+#elif defined(__arc__) || defined(__tilegx__)
+#elif defined(__arm__)
+#elif defined(__i386__) || defined(__x86_64__)
         { "lib64",    0, "usr/lib/x86_64-linux-gnu\0"
                          "usr/lib64\0",                "ld-linux-x86-64.so.2" },
+#  define HAVE_LIB64_DIRS 1
+#elif defined(__ia64__)
+#elif defined(__m68k__)
```

Comment:
so we know for some archs that they never had lib64/lib32 because there never was any other flavour of the CPU for a different register width. m68k is one of them, hence we can probably already skip that arch here.

This is interesting btw:

https://sourceware.org/glibc/wiki/ABIList
https://wiki.debian.org/Multiarch/LibraryPathOverview

According to those docs, it's pretty much these archs that need the symlink in some form for either lib32 or lib64: mips, x86/x86-64, ppc/ppc64, sparc.

Reply:
Those docs aren't complete apparently though, as #14311 suggests, which asks for this on aarch64 and s390x apparently. Maybe fedora actually insists on lib64 and thus /lib must be a symlink to it? yikes...

Reply:
So the ELF ABI specs dictate /lib64/ld-lsb-s390x.so.3 is the right name for ld.so on s390x:

https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwi07YbSxaH0AhVdSPEDHfo4AZ4QFnoECB0QAQ&url=https%3A%2F%2Frefspecs.linuxfoundation.org%2FLSB_3.2.0%2FLSB-Core-S390X%2FLSB-Core-S390X.pdf&usg=AOvVaw046SYJRRVtNCLApXjobCPX

Reply:
I added an exclude fro m68k now. I'm not confident enough to add any other excludes.

Reply:
Yeah, but that document is from 2007. After looking at the debian docs, which get the defines wrong (or at least different than what I see on an actual Fedora system), I'd be wary to add defines without a system to verify them against. People have access to s390x, so hopefully they'll fill this in. With the scaffolding in place, it should be fairly easy.

Reply:
OK, with the information in https://github.com/systemd/systemd/issues/14311, we can be pretty certain about the paths. I added a commit for this now.